resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/providers/sse/InboundSseEventImpl.java

```diff
@@ -97,9 +97,17 @@ public Builder write(byte[] data)
 
       public InboundSseEvent build()
       {
+         //from https://html.spec.whatwg.org/multipage/server-sent-events.html#processField
+         //If the data buffer's last character is a U+000A LINE FEED (LF) character, 
+         //then remove the last character from the data buffer
+         byte[] result = dataStream.toByteArray();
+         if (result.length > 0 && '\n' == result[result.length-1]) 
+         {
+            result = Arrays.copyOfRange(result, 0, result.length-1);
+         }
```

Comment:
Maybe it's not worth it, but a subclass of ByteArrayOutputStream could look at ByteArrayOutputStream.buf and decide to copy all of buf or all but the last byte, saving one copy.

Reply:
Oh, well, you already have EventByteArrayOutputStream ...

Reply:
@ronsigal  Thanks for the review and comments. You are right. We can use EventByteArrayOutputStream here too to avoid copy the array for 2 times.  I've fixed this . Please help review again.